### DISCUSSION TITLE
Add early 2000s / finish 90s hardware support

### DISCUSSION THREAD

**Discussion by SolDev69** [86Box/86Box#2690] [Ideas]
**Is your feature request related to a problem? Please describe.**
This feature request is not related to a problem, it would simply just expand the latest hardware currently supported due to GPU hard limit being slightly redundant.
**Describe the solution you'd like**
Currently the fastest GPU you can get is the Voodoo3 3000, and I get that newer hardware isn't supported due to modern hardware limitations but even GeForce 256 released in the same year as the Voodoo3 and it's not currently supported. I would love to see early 2000s GPUs supported as even my GeForce GTX 1650 Mobile is enough for Voodoo3 emulation (unless I am seeing signs of slowdown that I don't understand, usage seems to be fine in task manager), so more modern/higher end GPUs can probably emulate maybe up to the GeForce4/early Radeon series? I understand that there are hardware limits however as I nearly fried my Core i5-9300H trying to install Windows 98 with a Pentium II Overdrive/Xeon selected, however if higher end CPUs can emulate this fine enough for y'all to support it why not support some Pentium IIIs? I know y'all are probably going to flag this as too new of hardware but again I'm only asking for hardware that was released in 2002 at the latest. If GeForce4s are too new maybe then stick with GeForce 256? Idk I feel like the hardware emulation limit can be pushed slightly. Unless even the highest end systems today struggle with emulating the highest end config on here which they might as I don't have access to a high end system myself for testing.
**Describe alternatives you've considered**
Ideally this would have up to GeForce 6 and Pentium 4 support but again that's too new of hardware and even the highest end systems probably can't handle that. Maybe in like 10 or 20 years tho.

If this isn't possible adding somewhere to see the recommended hardware for certain configs would be nice so that people don't overestimate their systems and accidentally fry their PCs, and so that people stop asking for newer hardware support.

**Additional context**
I know that y'all get a bunch of these feature requests but most of them are asking for stupidly new hardware that would fry any system and I really hope this doesn't end up being one of these posts. I don't know much about emulation and as I said before I don't have a high end system to test higher end currently supported configurations so I don't know if this is feasible but I hope this is.

**Comment by MrPepka** [comment]
The reason no GeForce or even the Riva TNT series cards have been introduced is because the Riva and GeForce documentation is proprietary and not publicly available + some GeForce (like the GeForce 256 you mentioned) are too heavy to emulate (voodoo 3 3000 was less demanding)
As for the Pentium III emulation, the emulation of the Pentium II or even the first Pentium can be slowed down when the clock is set to 500 or more, moreover, in PCBox there is Pentium III emulation and it is inefficient

**Comment by re9177** [comment]
to add to the top, there also has to be interest from a contributor side of things to even maintain such a complex card like this, 3D cards are not easy to work with, so even if we had the datasheets, someone would still have to be interested to spend their time working on it too.

**Comment by dhrdlicka** [comment]
The [FAQ](http://86box.net/faq) has a section that answers your question about newer CPUs:

>### Are you going to add emulation of the Pentium 3 and/or newer CPUs?
>
>In short, no. Newer CPUs are way too powerful and even the top-end systems that are currently on the market are not nearly performant enough to be able to emulate them at usable speeds. In fact, we already had to add some low-clocked variants of the Pentium 2 that never actually existed, just so more people could use it!
>
>For further reading, team member RichardG wrote a [blog post](/2022/03/21/why-not-p3) that goes into the details of what makes the emulation of newer CPUs so controversial.
